providers/oidc.go

```diff
@@ -273,10 +279,27 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 	return claims, nil
 }
 
+func extractGroupsFromRawClaims(rawClaims map[string]interface{}, groupsClaim string) []string {
```

Comment:
Nice refactor 👍

Reply:
NIT: since we are passing in something from our `p *OIDCProvider` anyway as `groupsClaim` is this potentially more elegant as `func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {`

Reply:
Yup makes sense will do this!

> Thanks for bearing with us throughout all the back and forth. Looks good! Thanks for working on this.
> 
> Go ahead and add a CHANGELOG entry and I'll merge

All done! including fixing the nit, I added an important note that I think makes sense and that is that a user just ensures they have the `groups` scope configured and that they may be logged out intially if they did not have groups but will be authenticated upon doing the login dance.